### PR TITLE
Fix workspace assignment on screen change

### LINKED ISSUE #4261: since 4.19, workspaces don't move to configured output when it gets available
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

##### example config: `workspace 1 output DisplayPort-0`
When `DisplayPort-0` is not connected, `workspace 1` displays on a different output.
When `DisplayPort-0` gets connected, `workspace 1` doesn't move to output `DisplayPort-0` like it did pre 4.19.

## Expected Behavior

When the configured output gets available, the workspace should move to the output like in pre 4.19 versions.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version:  4.19
</pre>

<pre>
- Linux Distribution & Version: Arch Linux
- Are you using a compositor (e.g., xcompmgr or compton): xcompmgr
</pre>

### PR DESCRIPTION
**Pull request by abeutot**
Fixes #4261

### CODE REVIEW

File: src/randr.c [hunk 1]
```diff
@@ -440,27 +440,27 @@ void init_ws_for_output(Output *output) {
 
     /* Iterate over all workspaces and check if any of them should be assigned
      * to this output. */
-    Con *output_con;
-    TAILQ_FOREACH (output_con, &(croot->nodes_head), nodes) {
-        if (con_is_internal(output_con)) {
+    Con *workspace;
+    TAILQ_FOREACH (workspace, &all_cons, all_cons) {
```
Comment (orestisfl) [inline src/randr.c:447]: Maybe update the comment above to explain why we use this loop, future-proofing this change

**Review by orestisfl — APPROVED** [review]
Thanks :)

File: src/randr.c (hunk 1 above)
Comment (abeutot) [reply src/randr.c:447]: Hello. I added a comment. Let me know if it matches your expectations.

**Comment by abeutot** [comment]
Also I added an entry in the RELEASE-notes as required by the contribution guide. I don't know if my formatting is correct.

File: RELEASE-NOTES-next [hunk 2]
```diff
@@ -11,10 +11,13 @@ strongly encouraged to upgrade.
...
+  • i3-nagbar: position on focused monitor by default
+  • i3-nagbar: add option to position on primary monitor
 
  ┌────────────────────────────┐
  │ Bugfixes                   │
  └────────────────────────────┘
 
   • when initializing new outputs, avoid duplicating workspace numbers
+  • fix workspaces not moving to assigned output after said output is available
+    (regression introduced in 4.19)
```
Comment (orestisfl) [inline RELEASE-NOTES-next]:
```suggestion
  • fix workspaces not moving to assigned output after output becomes available
```

**Review by orestisfl — CHANGES_REQUESTED** [review]

**Comment by orestisfl** [comment]
Can you rebase with next and make everything one single commit please? The diff is broken right now.

**Comment by wlhlm** [comment]
As this fixes a regression, I think a testcase would be appropriate to include here so this won't happen again.

File: RELEASE-NOTES-next (hunk 2 above)
Comment (abeutot) [reply RELEASE-NOTES-next]: Done.

**Comment by abeutot** [comment]
> Can you rebase with next and make everything one single commit please? The diff is broken right now.

Done.

**Comment by orestisfl** [comment]
@wlhlm We don't have any randr tests unfortunately and I am not sure if it is possible to emulate reliably this in our test suite.

**Comment by orestisfl** [comment]
@abeutot there are still 6 commits to be merged here. One way to fix this is to get the sha of the commit (81c7b2dd66630ab515d91722ccba453c4517473a) `git reset --hard origin/next` and then `git cherry-pick 81c7b2dd66630ab515d91722ccba453c4517473a`.

**Comment by abeutot** [comment]
> @abeutot there are still 6 commits to be merged here. One way to fix this is to get the sha of the commit ([81c7b2d](https://github.com/i3/i3/commit/81c7b2dd66630ab515d91722ccba453c4517473a)) `git reset --hard origin/next` and then `git cherry-pick 81c7b2dd66630ab515d91722ccba453c4517473a`.

Oops. My bad. I forgot about the rebase part.

**Review by orestisfl — APPROVED** [review]
Thank you very much!

**Comment by orestisfl** [comment]
@stapelberg this is a regression fix

**Comment by orestisfl** [comment]
Introduced new label :)

**Comment by abeutot** [comment]
Thanks a lot. I didn't expect to get reviewed this fast :)

**Pull request merged**